plugins/hls-semantic-tokens-plugin/src/Ide/Plugin/SemanticTokens/Mappings.hs

```diff
@@ -32,30 +34,22 @@ import           Language.LSP.VFS                hiding (line)
 -- * 1. Mapping semantic token type to and from the LSP default token type.
 
 -- | map from haskell semantic token type to LSP default token type
-toLspTokenType :: HsSemanticTokenType -> SemanticTokenTypes
-toLspTokenType tk = case tk of
-  -- Function type variable
-  TFunction     -> SemanticTokenTypes_Function
-  -- None function type variable
-  TVariable     -> SemanticTokenTypes_Variable
-  TClass        -> SemanticTokenTypes_Class
-  TClassMethod  -> SemanticTokenTypes_Method
-  TTypeVariable -> SemanticTokenTypes_TypeParameter
-  -- normal data type is a tagged union type look like enum type
-  -- and a record is a product type like struct
-  -- but we don't distinguish them yet
-  TTypeCon      -> SemanticTokenTypes_Enum
-  TDataCon      -> SemanticTokenTypes_EnumMember
-  TRecField     -> SemanticTokenTypes_Property
-  -- pattern syn is like a limited version of macro of constructing a term
-  TPatternSyn   -> SemanticTokenTypes_Macro
-  -- saturated type
-  TTypeSyn      -> SemanticTokenTypes_Type
-  -- not sure if this is correct choice
-  TTypeFamily   -> SemanticTokenTypes_Interface
+toLspTokenType :: SemanticTokensConfig  -> HsSemanticTokenType -> SemanticTokenTypes
+toLspTokenType conf tk = case tk of
+  TFunction     -> runIdentity $ stFunction conf
+  TVariable     -> runIdentity $ stVariable conf
+  TClassMethod  -> runIdentity $ stClassMethod conf
+  TTypeVariable -> runIdentity $ stTypeVariable conf
+  TDataCon      -> runIdentity $ stDataCon conf
+  TClass        -> runIdentity $ stClass conf
+  TTypeCon      -> runIdentity $ stTypeCon conf
+  TTypeSyn      -> runIdentity $ stTypeSyn conf
+  TTypeFamily   -> runIdentity $ stTypeFamily conf
+  TRecField     -> runIdentity $ stRecField conf
+  TPatternSyn   -> runIdentity $ stPatternSyn conf
 
 lspTokenReverseMap :: Map.Map SemanticTokenTypes HsSemanticTokenType
```

Comment:
seems like this might need the config too? Or I guess this is only used in testing? I'd be inclined to make it explicit anyway

Reply:
Would it be better to take the config that you're using and error in that case? Since otherwise you could end up with misleading test output when you have customized the config?

Reply:
I consider it too, it won't be a confusion, because I have now parametrized the test output, for result with customized configuration, it won't convert the result back to hs semantic, but show the lap semantic token instead.

Reply:
`error in that case` does seems to be good though.

Reply:
done